bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -454,29 +501,33 @@ public void create(Composite parent, int style) {
 void browserDispose(Event event) {
 	containingEnvironment.instances.remove(this);
 
-	if (webView_2 != null) webView_2.Release();
 	if (environment2 != null) environment2.Release();
-	settings.Release();
-	webView.Release();
-	webView_2 = null;
+	if (settings != null) settings.Release();
+	lastWebViewTask.thenAccept(it -> {
+		it.Release();
+		if (!lastWebView_2Task.isCancelled()) lastWebView_2Task.join().Release();
```

Comment:
lastWebView_2Task is never null since its initialized in the beginning with a empty completable future. We rather check for if it is cancelled, which can only happen in setupBrowser().

Reply:
the assignment now happens inside the runnable in `thenAccept` so no NPE can occur